dao/src/main/java/org/thingsboard/server/dao/sqlts/insert/latest/psql/PsqlLatestInsertTsRepository.java

```diff
@@ -151,4 +161,10 @@ public int getBatchSize() {
             }
         });
     }
+
+    private static List<TsKvLatestEntity> sortEntities(List<TsKvLatestEntity> entities) {
```

Comment:
Insert of sorting in each saveOrUpdate, please introduce interface TbSqlQueueEntity that extends Comparable and highlight that the sorting should happen based on the primary key of SQL entity to avoid deadlocks in cluster mode.

Reply:
or path Comparator<E> to the init function of the queue.